packages/swapper/src/swappers/ChainflipSwapper/swapperApi/getTradeQuote.ts

```diff
@@ -1,366 +1,154 @@
-import type { AssetId } from '@shapeshiftoss/caip'
 import { CHAIN_NAMESPACE, fromAssetId, solAssetId } from '@shapeshiftoss/caip'
 import type { GetFeeDataInput } from '@shapeshiftoss/chain-adapters'
 import type { KnownChainIds } from '@shapeshiftoss/types'
-import type { Result } from '@sniptt/monads'
 import { Err, Ok } from '@sniptt/monads'
 import type { AxiosError } from 'axios'
-import { v4 as uuid } from 'uuid'
 
-import { getDefaultSlippageDecimalPercentageForSwapper } from '../../../constants'
-import type {
-  CommonTradeQuoteInput,
-  GetUtxoTradeQuoteInput,
-  SwapErrorRight,
-  SwapperDeps,
-  TradeQuote,
-} from '../../../types'
+import type { QuoteFeeData } from '../../../types'
 import {
-  type GetEvmTradeQuoteInput,
-  type ProtocolFee,
-  SwapperName,
+  type CommonTradeQuoteInput,
+  type SwapperDeps,
   TradeQuoteError,
+  type TradeQuoteResult,
 } from '../../../types'
-import { getRate, makeSwapErrorRight } from '../../../utils'
+import { makeSwapErrorRight } from '../../../utils'
+import { CHAINFLIP_BAAS_COMMISSION } from '../constants'
 import {
-  CHAINFLIP_BAAS_COMMISSION,
-  CHAINFLIP_BOOST_SWAP_SOURCE,
-  CHAINFLIP_DCA_BOOST_SWAP_SOURCE,
-  CHAINFLIP_DCA_QUOTE,
-  CHAINFLIP_DCA_SWAP_SOURCE,
-  CHAINFLIP_REGULAR_QUOTE,
-  CHAINFLIP_SWAP_SOURCE,
-  usdcAsset,
-} from '../constants'
-import type { ChainflipBaasQuoteQuote, ChainflipBaasQuoteQuoteFee } from '../models'
-import { chainflipService } from '../utils/chainflipService'
-import { getEvmTxFees } from '../utils/getEvmTxFees'
-import { getUtxoTxFees } from '../utils/getUtxoTxFees'
-import { getChainFlipIdFromAssetId, isSupportedAssetId, isSupportedChainId } from '../utils/helpers'
+  calculateChainflipMinPrice,
+  getChainFlipIdFromAssetId,
+  getChainFlipSwap,
+} from '../utils/helpers'
+import { getRateOrQuote } from '../utils/rate-quotes'
 
-export const _getTradeQuote = async (
+export const getTradeQuote = async (
   input: CommonTradeQuoteInput,
   deps: SwapperDeps,
-): Promise<Result<TradeQuote[], SwapErrorRight>> => {
+): Promise<TradeQuoteResult> => {
   const {
-    sellAsset,
-    buyAsset,
     accountNumber,
+    sendAddress,
     receiveAddress,
     sellAmountIncludingProtocolFeesCryptoBaseUnit: sellAmount,
-    affiliateBps: commissionBps,
   } = input
 
-  if (!isSupportedChainId(sellAsset.chainId)) {
```

Comment:
Even though this is a trade quote, let's keep all these in their original place and add additional checks below 🙏🏽